resources/definitions/bcn3dprinter.def.json

```diff
@@ -0,0 +1,349 @@
+            },
+            "smart_purge_slope": {
+              "label": "Increase Distance by",
+              "description": "Distance to purge per minute idle. Purged distance will increase this amount each minute the toolhead remains idle. The real purged distance will never be shorter than the Minimum Purge Distance, or greater than the Maximum Purge Distance.",
+              "unit": "mm/min",
+              "type": "float",
+              "default_value": 0,
+              "minimum_value": "0",
+              "minimum_value_warning": "0",
+              "maximum_value_warning": "50",
+              "enabled": "smart_purge and purge_in_bucket and idex_print_mode == 'standard'",
+              "settable_per_extruder": true,
+              "comments": "BCN3D setting"
+            },
+            "smart_purge_slope_gcode": {
+              "label": "Distance Gradient",
+              "description": "GCode to send proper value to the machine.",
+              "unit": "mm/s",
+              "type": "float",
+              "default_value": 0,
+              "value": "smart_purge_slope / float(60)",
+              "enabled": false,
+              "settable_per_extruder": true,
+              "comments": "BCN3D setting"
+            }
+          }
+        },
+        "smart_purge_enable_gcode": {
+          "label": "Smart Purge Enable GCode",
+          "description": "GCode to enable smart purge commands.",
+          "type": "enum",
+          "options": {
+            ";": "No",
+            "": "Yes"
+          },
+          "default_value": "",
+          "value": "'' if smart_purge and purge_in_bucket else ';'",
+          "enabled": false,
+          "settable_per_extruder": true,
+          "comments": "BCN3D setting"
+        },
+        "prime_tower_closest_to": {
+          "label": "Prime Tower Preferred Position",
+          "description": "Select the preferred area of the buildplate to automatically place the prime tower.",
+          "type": "enum",
+          "options": {
+            "top-left": "Top Left",
+            "top": "Top",
+            "top-right": "Top Right",
+            "right": "Right",
+            "bottom-right": "Bottom Right",
+            "bottom": "Bottom",
+            "bottom-left": "Bottom Left",
+            "left": "Left"
+          },
+          "default_value": "top-right",
+          "enabled": "resolveOrValue('prime_tower_enable')",
+          "settable_per_mesh": false,
+          "settable_per_extruder": false,
+          "comments": "BCN3D setting"
+        }
+      }
+    },
+    "blackmagic": {
+      "children": {
+        "material_type": {
+          "label": "Material type",
+          "description": "Type of material, used internally within BCN3D qualities.",
+          "type": "str",
+          "default_value": "material",
+          "enabled": false,
+          "settable_per_extruder": true,
+          "comments": "BCN3D setting"
+        },
+        "machine_nozzle_type": {

```

Comment:
The only time this or `material_type` is used is in a comment in the generated GCODE. Is that why you are recommending `startslicejob` since it can add Python-derived replacement tokens based on the current material or nozzle?